app/models/provider.rb

```diff
@@ -29,6 +30,14 @@ def <=>(other)
     self.Miles <=> other.Miles
   end
 
+  def hexdigest
+    Digest::MD5.hexdigest(attributes.except(:ProviderHexdigest).to_a.join('|'))
```

Comment:
Copied from Slack:

Could we use a SHA digest instead of MD5? I’m not really up-to-speed on the performance constraints.

In newer Ruby versions, SHA-1 is as fast or faster than MD5. And SHA-256 is only 2x slower: https://gist.github.com/morimori/1330095